sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,164 +1,122 @@
-# Azure.Core public serialization samples
+# Azure.Core public serialization
 
-## Using explicit cast
+The .NET Azure SDK libraries support serialization and deserialization of most client library model types using the static `ModelSerializer` class. 
 
-When using protocol methods for advanced handling of RequestContext it is still possible to use the strongly typed models.
-There is an explicit cast operator that can be used to convert the protocol Response to the strongly typed model.
-There is also an explicit cast operator that can be used to convert the strongly typed model to the protocol RequestContent.
+## Key Concepts
 
-### Serialization
+- [Using the ModelSerializer](#using-the-modelserializer)
+- [Using models with protocol methods](#using-models-with-protocol-methods)
+- [Using JsonSerializer](#using-jsonserializer)
+- [Envelope bring your own model case](#envelope-bring-your-own-model-case)
 
-```C# Snippet:ExplicitCast_Serialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-DogListProperty dog = new DogListProperty("myPet");
-Response response = client.CreatePet("myPet", (RequestContent)dog);
-var response2 = client.CreatePet("myPet", RequestContent.Create(dog));
-```
-
-### Deserialization
-
-```C# Snippet:ExplicitCast_Deserialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-Response response = client.GetPet("myPet");
-DogListProperty dog = (DogListProperty)response;
-Console.WriteLine(dog.IsHungry);
-```
+## Using the ModelSerializer
+The `ModelSerializer` class enables users to serialize and deserialize any Azure models that implement the `IModelSerializable<T>` interface. The following example shows a Dog model being serialized and deserialized.
 
-Given that explicit cast does not allow for serialization options we might also consider a static `FromResponse` and instance `ToRequestContent` methods.
+```C# Snippet:BaseModelSerializer
+Dog doggo = new Dog
+{
+    Name = "Doggo",
+    Age = 7
+};
 
-## Using ModelSerializer for NewtonSoftJson
+BinaryData data = ModelSerializer.Serialize(doggo);
 
-By using the ModelSerializer class, a new instance of Dog does not need to be created before calling Deserialize. Also added ObjectSerializer to Options class so different kinds of Serializers can be used.
+Dog dog = ModelSerializer.Deserialize<Dog>(data);
+```
 
-### Serialization
+By default, the ModelSerializer gives you the Json representation of the model. Some services accept XML so if you want to send the serialized data to an Azure service that accepts XML, you can use the `ModelSerializerFormat` enum to specify the format. By using the `Wire` format, the serializer automatically picks what the service would use.
```

Comment:
Is `Wire` only useful for services that use XML, or is there a difference between the JSON format as well?  

For example, if I don't use `Wire` and I take my round-trip model, serialize it, store it for a bit, and then send it to my UI in the browser with the JS client library submitting it to the service - does that work or do I need to ensure `Wire`?

What is the "other" format that we use as the default?  Should we touch briefly on which scenarios each is intended to be used for?